mPowerSDK/Dashboard/APHCorrelationsSelectorViewController.m

```diff
@@ -0,0 +1,203 @@
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+#import "APHCorrelationsSelectorViewController.h"
+#import "APHLocalization.h"
+#import "APHScoring.h"
+
+#import <APCAppCore/UIColor+APCAppearance.h>
+
+@interface APHCorrelationsSelectorViewController ()
+@property (strong, nonatomic) APHScoring *scoring;
+@property (strong, nonatomic) NSArray *scoringObjects;
+
+@property (strong, nonatomic) UITableView *tableView;
+@property (nonatomic) CGRect downFrame;
+@property (nonatomic) CGRect upFrame;
+@end
+
+@implementation APHCorrelationsSelectorViewController 
+
+- (id)initWithScoringObjects:(NSArray *)scoringObjects
+{
+    self = [super init];
+    if (self) {
+        self.view.backgroundColor = [[UIColor blackColor] colorWithAlphaComponent:0.8];
+        self.scoringObjects = scoringObjects;
+        
+        [self initTableView];
+        [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"Cell"];
+    }
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+}
+
+- (void)viewWillAppear:(BOOL)__unused animated
+{
+    [super viewWillAppear:animated];
+    
+    [self.tableView reloadData];
+    [self moveUpTableView];
+}
+
+- (void)viewWillDisappear:(BOOL)animated
+{
+    [super viewWillDisappear:animated];
+}
+
+#pragma mark - Helper Functions
+
+- (void)dismissView
+{
+    [self moveDownTableView];
+}
+
+- (void)initTableView
+{
+    self.downFrame = CGRectMake(0, self.view.frame.size.height, self.view.frame.size.width, self.view.frame.size.height - 20);
+    self.upFrame = CGRectMake(0, 20, self.view.frame.size.width, self.view.frame.size.height - 20);
+    
+    self.tableView = [[UITableView alloc] initWithFrame:self.downFrame style:UITableViewStylePlain];
+    self.tableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    [self.view addSubview:self.tableView];
+}
+
+- (void)moveUpTableView
+{
+    [UIView
+     animateWithDuration:0.5
+     animations:^{
+         self.tableView.frame = self.upFrame;
+     }];
+}
+
+- (void)moveDownTableView
```

Comment:
No. if this is *only* called from `-dismissView` and you want to have it dismiss the view controller when it's done then use a completion that is sent into this method. This method is asking for future maintenance nightmares to the poor devs who inherit it.